pinecone/index_connection_test.go

```diff
@@ -0,0 +1,161 @@
+package pinecone
+
+import (
+	"context"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
```

Comment:
Read up a bit on the testing framework in case anyone else is curious: https://github.com/stretchr/testify.

This was the one you mentioned was controversial? Seems fine to me, although I'm not sure if that's something we need to consider at this point.

Also less relevant now, but should we consider having a specific folder path for test files?

Reply:
Yeah, more that using anything not built in is controversial. This package allows for things like setup and tear-down on suite, sub-suite, test levels. It's just more like other testing frameworks folks are used to. Seemed like a good addition.